### PR TITLE
Implement Account & Setting UI screen

### LINKED ISSUE #31: Implement Account & Setting UI screen

### PR DESCRIPTION
**Pull request by thanhdo29**
## Related issues
Resolve #31 

## Summary
implement account & setting ui screen

## How to test
run app

## Screenshots
homeui

### CODE REVIEW

**Comment by NguyenVanViet2004** [comment]
image

**Comment by NguyenVanViet2004** [comment]
Simulator Screenshot - iPhone 15 Pro Max - 2024-07-28 at 23 27 59

**Comment by NguyenVanViet2004** [comment]
image

**Comment by NguyenVanViet2004** [comment]
image

**Comment by NguyenVanViet2004** [comment]
image

File: components/templates/AccountTemplate.tsx [hunk 1]
```diff
@@ -1,22 +1,99 @@
...
-import { Text } from 'tamagui'
+import { StyleSheet, useColorScheme } from 'react-native'
+import { ScrollView, Text, View } from 'tamagui'
+
+import SafeArea from '~/components/atoms/SafeArea'
+import { ChooseMethod } from '~/components/molecules/ChooseMethod'
+import Header from '~/components/molecules/common/Header'
+import getColors from '~/constants/Colors'
+import useTranslation from '~/hooks/useTranslation'
 const AccountScreen: React.FC = () => {
```
Comment (NguyenVanViet2004) [inline components/templates/AccountTemplate.tsx:11]: `AccountTemplate`

File: components/templates/AccountTemplate.tsx [hunk 2]
```diff
@@ -1,22 +1,99 @@
...
+          backgroundColor={colors.lightSilver}>
+          <Header
+            leftIcon={
+              <AntDesign name="left" size={15} color={colors.midnightBlue} />}
+            label="Account & Settings" />
+
+          <Text
+            fontSize={18}
+            fontWeight={500}
+            marginVertical={24}>{t('account.account')}</Text>
```
Comment (NguyenVanViet2004) [inline components/templates/AccountTemplate.tsx:30]: Why don't you use the title of `Header component`?

File: components/molecules/common/Header.tsx [hunk 3]
```diff
@@ -10,44 +10,54 @@ interface props {
...
+    const colors = getColors(useColorScheme())
+    const renderIcon = (icon: number | ReactElement): React.ReactElement => {
+      if (isNumber(icon)) {
+        return <Image source={icon} />
+      }
+      return <Button
+        unstyled
+        padding={10}
+        borderRadius={50}
+        backgroundColor={colors.white}
```
Comment (NguyenVanViet2004) [inline components/molecules/common/Header.tsx:27]: is `backgroundColor` necessary?

File: components/templates/AccountTemplate.tsx [hunk 4]
```diff
@@ -1,22 +1,99 @@
...
+
+          <ChooseMethod
```
Comment (NguyenVanViet2004) [inline components/templates/AccountTemplate.tsx:32]: I think you should use `map` here, what do you think?

File: locales/en.json [hunk 5]
```diff
@@ -20,5 +20,17 @@
         "topSeller": "Top Sell",
         "bestChoice": "BEST CHOICE",
         "bestSeller": "Best Seller"
+    },
+    "account":{
+        "notificationSetting":"Notification Setting",
+        "shippingAddress":"Shipping Address",
+        "paymentInfo":"Payment Info",
+        "deleteAccount":"Delete Account",
+        "eneble-FaceID-For-LogIn":"Eneble Face ID For Log In",
```
Comment (NguyenVanViet2004) [inline locales/en.json:29]: use `Camel Case`

File: components/molecules/ChooseMethod.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  const { t } = useTranslation()
+  const [isChecked, setIsChecked] = useState(false)
+  const colors = getColors(useColorScheme())
+  const renderIcon = (icon: React.ReactElement): React.ReactElement => {
+    return icon
+  }
+
+  const handleSwitchChange = (checked: boolean): void => {
+    setIsChecked(checked)
+    props.onCheckedChange?.(checked)
```
Comment (NguyenVanViet2004) [inline components/molecules/ChooseMethod.tsx:26]: I do not understand this line, can you explain it?

File: components/molecules/common/Header.tsx (hunk 3 above)
Comment (NguyenVanViet2004) [reply components/molecules/common/Header.tsx:29]: what about this comment?

File: components/templates/AccountTemplate.tsx [hunk 7]
```diff
@@ -1,23 +1,84 @@
...
+import getColors from '~/constants/Colors'
+import dataMethodAccount from '~/constants/DataMethodAccount'
+import dataMethodSetting from '~/constants/DataMethodSetting'
+import useTranslation from '~/hooks/useTranslation'
+const AccountTemplate: React.FC = () => {
+  const { t } = useTranslation()
+  const colors = getColors(useColorScheme())
+
+  const renderIcon = (type: 'MaterialCommunityIcons' | 'AntDesign' | 'Ionicons',
+    name: any, size: number, color: any): React.ReactElement => {
```
Comment (NguyenVanViet2004) [inline components/templates/AccountTemplate.tsx:18]: why `name` is any?

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Please fix all bugs

File: components/molecules/ChooseMethod.tsx (hunk 6 above)
Comment (thanhdo29) [reply components/molecules/ChooseMethod.tsx:26]: - setIsChecked() is used to update the switch's state if it changes
- if Switch changes then props.onCheckedChange?.(checked) will be called to propagate the current state to the AccountTemeplate

File: components/templates/AccountTemplate.tsx [hunk 8]
```diff
@@ -14,17 +14,39 @@ const AccountTemplate: React.FC = () => {
   const { t } = useTranslation()
   const colors = getColors(useColorScheme())
 
-  const renderIcon = (type: 'MaterialCommunityIcons' | 'AntDesign' | 'Ionicons',
-    name: any, size: number, color: any): React.ReactElement => {
+  type IconType = 'MaterialCommunityIcons' | 'AntDesign' | 'Ionicons'
```
Comment (NguyenVanViet2004) [inline components/templates/AccountTemplate.tsx:17]: If I want to pass a different data type, how can I do that?

File: components/molecules/common/Header.tsx [hunk 9]
```diff
@@ -4,50 +4,65 @@ import { useColorScheme } from 'react-native'
...
 
 import getColors from '~/constants/Colors'
+import useTranslation from '~/hooks/useTranslation'
 
 interface props {
   leftIcon?: number | React.ReactElement
   rightIcon?: number | React.ReactElement
   title?: string
   subtitle?: string
+  label?: string
```
Comment (NguyenVanViet2004) [inline components/molecules/common/Header.tsx:14]: Why don't you use `title`?

File: components/templates/AccountTemplate.tsx [hunk 10]
```diff
@@ -1,23 +1,85 @@
...
+const AccountTemplate: React.FC = () => {
+  const { t } = useTranslation()
+  const colors = getColors(useColorScheme())
+
+  const handleSwitchChange = (checked: boolean): void => {
+    console.log('Switch state :', checked)
+  }
+
+  const renderIcon = (
+    IconComponent: any,
```
Comment (NguyenVanViet2004) [inline components/templates/AccountTemplate.tsx:22]: it should be in the `atoms` folder

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

**Comment by NguyenVanViet2004** [comment]
|--|--|
|---|---|
|image|image|

**Comment by NguyenVanViet2004** [comment]
https://github.com/NguyenVanViet2004/FE-Shoes-Ecommerce/pull/33/files#r1694280303

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**